test/utils/probability/test_lin_ess.py

```diff
@@ -462,3 +462,44 @@ def test_multivariate(self):
             self.assertEqual(X_high_d.shape, torch.Size([16, d]))
             self.assertTrue(sampler._is_feasible(X_high_d.T).all())
             self.assertEqual(sampler.lifetime_samples, num_samples)
+
+    def test_batch_mcmc(self):
+        d = 5
+        for i in range(2):
+            torch.manual_seed(i)
+
+            for dtype, atol in zip((torch.float, torch.double), (2e-5, 1e-12)):
+                tkwargs = {"device": self.device, "dtype": dtype}
+
+                # special case: N(0, I) truncated by a symmetric box
+                bounds = torch.cat(
+                    [-1 * torch.ones(1, d, **tkwargs), torch.ones(1, d, **tkwargs)],
+                    dim=0,
+                )
+
+                # Run a single Markov chain.
+                sampler = LinearEllipticalSliceSampler(
+                    bounds=bounds,
+                    check_feasibility=True,
+                    burnin=100,
+                )
+                samples = sampler.draw(n=100)
+                self.assertEqual(samples.shape, torch.Size([100, d]))
+
+                # Run 100 Markov chains.
+                batch_sampler = LinearEllipticalSliceSampler(
+                    bounds=bounds,
+                    check_feasibility=True,
+                    burnin=100,
+                    chains=100,
+                )
+                batch_samples = batch_sampler.draw(n=100)
+                self.assertEqual(batch_samples.shape, torch.Size([100 * 100, d]))
```

Comment:
About how long does this take to run on a CPU? We'll want tests to run within a second or two (not counting any slow imports) on any reasonable infra

Reply:
On a CPU, this code block takes about 1.5 second.

After removing the for loop as suggested in this [comment](https://github.com/pytorch/botorch/pull/2426#discussion_r1673846160
), the entire test file `test_lin_ess.py` runs in 2 second.

Is that a reasonable running time? We can always reduce the number of MCMC steps or the number of chains if needed.

Reply:
2 seconds is okay, but if we can make it faster without making the tests less informative, I'd vote to do that.

Reply:
I am reducing the problem size to `burnin=50`, `num_chains=50`, and `n=10`. Running a few hundred MCMC steps should be fast on any reasonable machines.

Now the entire test takes about 1.6 seconds. Note that the rest of tests already takes 1.5 seconds. Will push the change shortly.